File: AspNetMvcUnleashed/0672329980_source/Chapter11/VB/MvcApplication1/MvcApplication1/Controllers/VaryCustomController.vb
Public Class VaryCustomController
    Inherits Controller

    <OutputCache(Duration:=9999, VaryByParam:="None", VaryByCustom:="JS")> _
    Public Function Index() As ActionResult
        If Request.Browser.EcmaScriptVersion.Major > 0 Then
            Return View("IndexJS")
        End If

        Return View("Index")

    End Function

End Class

File: AspNetMvcUnleashed/0672329980_source/Chapter11/CS/MvcApplication1/MvcApplication1/Controllers/VaryCustomController.cs
using System.Web.Mvc;

namespace MvcApplication1.Controllers
{
    public class VaryCustomController : Controller
    {
        [OutputCache(Duration=9999, VaryByParam="None", VaryByCustom="JS")]
        public ActionResult Index()
        {
            if (Request.Browser.EcmaScriptVersion.Major > 0) 
                return View("IndexJS");
        
            return View("Index");

        }

    }
}
